src/lib/io/sol-gpio-impl-contiki.c

```diff
@@ -61,8 +61,10 @@ static void
 event_handler_cb(void *user_data, process_event_t ev, process_data_t ev_data)
 {
     struct sol_gpio *gpio = user_data;
+    bool val;
 
-    gpio->irq.cb(gpio->irq.data, gpio);
+    val = sol_gpio_read(gpio);
```

Comment:
just to check, do we need to read it here? Can't we put that in the ISR and save the value in `ev` or `ev_data`?